tket/src/Transformations/include/Transformations/SingleQubitSquash.hpp

```diff
@@ -85,35 +85,50 @@ class SingleQubitSquash {
    * @brief Construct a new Single Qubit Squash object.
    *
    * @param squasher The Squasher instance.
-   * @param reversed whether squashing is made back-to-front or front to back.
+   * @param reversed Whether squashing is made back to front or front to back
+   *      (default: false).
+   * @param circ The circuit to be squashed (optional).
    */
   SingleQubitSquash(
       std::unique_ptr<AbstractSquasher> squasher, bool reversed = false)
       : squasher_(std::move(squasher)),
         reversed_(reversed),
         circ_ptr_(nullptr) {}
+  SingleQubitSquash(
+      std::unique_ptr<AbstractSquasher> squasher, bool reversed = false,
+      Circuit &circ)
+      : squasher_(std::move(squasher)), reversed_(reversed), circ_ptr_(&circ) {}
 
   /**
    * @brief Squash an entire circuit, one qubit at a time.
    *
-   * @param circ The circuit to be squashed.
+   * Either pass a circuit as argument or use the current circuit.
```

Comment:
(Sorry, when I wrote "pass" I meant "class"...)
It still seems a bit weird to me. Instead of passing an argument to this method overriding the circuit stored in the class, how about just constructing a new class with the new circuit?

Reply:
Yes, you're right. I've changed this as you suggest.